app/src/main/java/com/blaizmiko/popcornapp/ui/fragments/movies/MovieDetailsFragment.java

```diff
@@ -0,0 +1,164 @@
+import com.google.android.flexbox.FlexWrap;
+import com.google.android.flexbox.FlexboxLayoutManager;
+import com.iarcuschin.simpleratingbar.SimpleRatingBar;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+
+import butterknife.BindView;
+
+public class MovieDetailsFragment extends BaseMvpFragment implements View.OnClickListener, LoadProgressView, MovieDetailsView {
+
+    public static MovieDetailsFragment newInstance() {
+        return new MovieDetailsFragment();
+    }
+
+    @InjectPresenter
+    DetailMoviesPresenter mMovieDetailsPresenter;
+
+    @InjectPresenter
+    LoadProgressPresenter mLoadProgressPresenter;
+
+    private GenresTagsAdapter mGenresTagsAdapter;
+    private int mMovieId;
+    private final int mStoryLineTextViewLinesMin = 3;
+    private final int mStoryLineTextViewLinesMax = 8;
+    private boolean mIsStoryLineTextViewOpen = false;
+    public static final String mBundleArgumentId = "ID";
+
+    //Bind views
+    @BindView(R.id.fragment_detail_movie_backdrop_image_view)
+    ImageView mBackdropImageView;
+    @BindView(R.id.fragment_detail_movie_poster_image_view)
+    ImageView mPosterImageView;
+    @BindView(R.id.fragment_detail_movie_rating_text_view)
+    TextView mRatingTextView;
+    @BindView(R.id.fragment_detail_movie_title_text_view)
+    TextView mTitleTextView;
+    @BindView(R.id.fragment_detail_movie_story_line_text_view)
+    TextView mStoryLineTextView;
+    @BindView(R.id.fragment_detail_movie_rating_bar)
+    SimpleRatingBar mRatingBar;
+    @BindView(R.id.fragment_detail_movie_genre_tags_recycler_view)
+    RecyclerView mGenresRecyclerView;
+    @BindView(R.id.fragment_detail_movie_short_info_linear_layout)
+    LinearLayout mShortInfoLinearLayout;
+    @BindView(R.id.fragment_detail_movie_progress_bar)
+    ProgressBar mProgressBar;
+
+    //Life cycle
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
+        mMovieId = getArguments().getInt(mBundleArgumentId);
+        return inflater.inflate(R.layout.fragment_detail_movies, container, false);
+    }
+
+    @Override
+    protected void bindViews() {
+        mMovieDetailsPresenter.loadMovie(mMovieId);
+        final FlexboxLayoutManager flexboxLayoutManager = new FlexboxLayoutManager(FlexDirection.ROW, FlexWrap.WRAP);
+        mGenresRecyclerView.setLayoutManager(flexboxLayoutManager);
+        mGenresTagsAdapter = new GenresTagsAdapter();
+        mGenresRecyclerView.setAdapter(mGenresTagsAdapter);
+    }
+
+    @Override
+    public void setMovie(Movie movie) {
+        mRatingTextView.setText(Double.toString(movie.getVoteAverage()));
+        mTitleTextView.setText(movie.getTitle());
+        mStoryLineTextView.setText(movie.getOverview());
+        mRatingBar.setRating(Float.parseFloat(new DecimalFormat(FormatUtils.ONE_DECIMAL, new DecimalFormatSymbols(Locale.US)).format(movie.getVoteAverage() / 2)));
```

Comment:
so, mb we need separate method in special class for this, and after that reuse this method